### PR TITLE
Initial switch to ts lib starter setup

### PR DESCRIPTION
**Pull request by iwburns**
setting this up to see if this travis build works

### CODE REVIEW

**Comment by iwburns** [comment]
So, this seems to be working well, however, I still need to make sure that publishing is going to work the way it used to (and if not, figure out what tweaks need to be made).

Gonna have to put this down for now, but it seems this is moving in the right direction!

**Comment by iwburns** [comment]
@brycehipp @ryanguill take a look if/when you guys have time and let me know what you think.

I _do_ still need to dig in a bit on the publishing side of things and make sure this will work properly, but I think I'm done with the "real" code changes here.  I may just bump the version on this to 0.7.0 and publish it to see if it works the way I think it will.

**Comment by iwburns** [comment]
I've made some tweaks here and I believe we're good to go now.  Take a look when you guys have time and let me know if you see anything that needs to be addressed.

**Comment by iwburns** [comment]
Ok, I'm ready for real this time!

**Comment by brycehipp** [comment]
from what I understand, everything looks great and is sound

**Comment by iwburns** [comment]
:tada: thanks!  I will merge and publish 0.7.0 tonight at some point.

@ryanguill please still feel free to take a look if you have time

File: test/util.ts [hunk 1]
```diff
@@ -1,116 +1,70 @@
...
-    .to.have.property('filter').that.is.a('function');
-  expect(val)
-    .to.have.property('forEach').that.is.a('function');
-  expect(val)
-    .to.have.property('equals').that.is.a('function');
-  expect(val)
-    .to.have.property('hasValue').that.is.a('function');
-  expect(val)
-    .to.have.property('contains').that.is.a('function');
+  expect(val).toHaveProperty('isSome'); expect(val.isSome).toBeInstanceOf(Function);
```
Comment (ryanguill) [inline test/util.ts]: do you intend for these to be on the same line? can you not chain the expectations with jest?

Comment (iwburns) [reply test/util.ts]: I did mean for them to be on the same line just for visibility's sake.  I agree it's not the prettiest code.  I tried chaining, but I couldn't quite find what I needed, and Jest's docs weren't very verbose, so I just went with this for now.

**Comment by iwburns** [comment]
I felt weird about expect not being imported as well.  I don't remember the exact reason I went with that, but I remember thinking it was odd.  I _think_ `expect()` is global somehow when you test files with `jest` but I'm not 100% sure how it works.

We can definitely modify the way that works, I just went with it because it worked for now.

**Comment by iwburns** [comment]
So, with that last modification (a tweak to .npmignore) coveralls isn't reporting back here with test results.

However, if you follow the Travis link, and look at the output (not sure if you guys can see that info or not) it says 100% of files, statements, branches, etc. are covered.

So, I'm going to ignore the fact that coveralls hasn't yet responded here and just merge and publish and see how this goes.

**Pull request merged**